src/index.ts

```diff
@@ -287,15 +288,28 @@ const useAsyncInternal = <R = UnknownResult, Args extends any[] = UnknownArgs>(
     }
   }, params);
 
-  return {
-    ...AsyncState.value,
-    set: AsyncState.set,
-    merge: AsyncState.merge,
-    reset: AsyncState.reset,
-    execute: executeAsyncOperation,
-    currentPromise: CurrentPromise.get(),
-    currentParams,
-  };
+  const currentPromise = CurrentPromise.get();
+
+  return useMemo(
+    () => ({
+      ...AsyncState.value,
+      set: AsyncState.set,
+      merge: AsyncState.merge,
+      reset: AsyncState.reset,
+      execute: executeAsyncOperation,
+      currentPromise,
+      currentParams,
+    }),
+    [
+      AsyncState.value,
+      AsyncState.set,
+      AsyncState.merge,
+      AsyncState.reset,
+      executeAsyncOperation,
+      currentPromise,
+      currentParams,
+    ]
+  );
```

Comment:
I noticed that `AsyncState.value` is never mutated, so the memo dependency is just `.value` itself.